ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -770,117 +777,118 @@ static std::condition_variable compile_count_cond;
 static void ggml_vk_create_pipeline_func(vk_device& device, vk_pipeline& pipeline, const std::string name, size_t spv_size, const void* spv_data, const std::string entrypoint,
                                          uint32_t parameter_count, uint32_t push_constant_size, std::array<uint32_t, 3> wg_denoms, std::vector<uint32_t> specialization_constants,
                                          uint32_t align, bool disable_robustness, bool require_full_subgroups, uint32_t required_subgroup_size) {
-    VK_LOG_DEBUG("ggml_vk_create_pipeline(" << device->name << ", " << name << ", " << entrypoint << ", " << parameter_count << ", " << push_constant_size <<
-                 ", (" << wg_denoms[0] << "," << wg_denoms[1] << "," << wg_denoms[2] << "), specialization_constants, " << align <<
-                 ", " << disable_robustness << ", " << require_full_subgroups << ", " << required_subgroup_size << ")");
     GGML_ASSERT(parameter_count > 0);
     GGML_ASSERT(wg_denoms[0] > 0 && wg_denoms[1] > 0 && wg_denoms[2] > 0); // NOLINT
 
-    pipeline = std::make_shared<vk_pipeline_struct>();
-    pipeline->name = name;
-    pipeline->parameter_count = parameter_count;
-    pipeline->push_constant_size = push_constant_size;
-    pipeline->wg_denoms = wg_denoms;
-    pipeline->align = align;
+    if (!pipeline) {
+        pipeline = std::make_shared<vk_pipeline_struct>();
+        pipeline->name = name;
+        pipeline->parameter_count = parameter_count;
+        pipeline->push_constant_size = push_constant_size;
+        pipeline->wg_denoms = wg_denoms;
+        pipeline->align = align;
+    }
 
-    vk::ShaderModuleCreateInfo shader_module_create_info({}, spv_size, reinterpret_cast<const uint32_t *>(spv_data));
-    pipeline->shader_module = device->device.createShaderModule(shader_module_create_info);
+    if (pipeline->needed && !pipeline->compiled) {
 
-    std::vector<vk::DescriptorSetLayoutBinding> dsl_binding;
```

Comment:
These diffs don't display well due to an indentation change. I suggest reviewing in a different diff tool.

Reply:
(gone in next revision)